### ISSUE TITLE
https://chaturbate.com/ JavaScript freezes

### ISSUE THREAD

**Issue by rami123** [MrAlex94/Waterfox-Old#45]
https://chaturbate.com/
freezes in clean Firefox classic install version 2021.11

I get message:
"a web page is slowing down your browser"

**Comment by Killogy** [comment]
If you use block Origin block gstatic.com. 

https://i.imgur.com/Mr2ctmn.jpg

**Comment by hawkeye116477** [comment]
> Firefox

So you're writing in wrong place. This isn't Mozilla's forum.

**Comment by Killogy** [comment]
> > Firefox
> 
> So you're writing in wrong place. This isn't Mozilla's forum.

I'm sure he meant Watefox Classic 2021.11. This issue is only occurring under Waterfox Classic, under Waterfox G4 this issue is not reproduced.

**Comment by rami123** [comment]
Yes I meant Watefox Classic 2021.11, not FireFox or Waterfox G4.

**Comment by rami123** [comment]
> If you use block Origin block gstatic.com.
> 
> https://i.imgur.com/Mr2ctmn.jpg

@Killogy  
I have blocked gstatic.com, but it does not help.

**Comment by Killogy** [comment]
> > If you use block Origin block [gstatic.com](http://gstatic.com).
> > https://i.imgur.com/Mr2ctmn.jpg
> 
> @Killogy I have blocked [gstatic.com](http://gstatic.com), but it does not help.

For me blocking only gstatic.com fixes this problem.

Blocking www.gstatic.com or www.google.com or google.com also fixes this problem, so test blocking some of those domains.

**Comment by Squall-Leonhart** [comment]
you have to disable the explicit filter on gstatic.com to be able to block it per domain.

**Comment by rami123** [comment]
Just a quick update, chatrubate is not that important, but I tried to understand better what is causing sites not to load. At least in this case when I whitelisted gstatic.com in ublock, chatrubate become slightly more responsive so ublock log could be used to see what runs. After I analyzed the log, I added this rule to uBlock My Filters and it solved the problem:
||www.gstatic.com/*.js$script,domain=chaturbate.com,important

This says that every java script that run on chaturbate.com and is originated from www.gstatic.com will be blocked. (Important tag at the end gives this rule priority over other rules)

Like many of you might know, other sites with problems like GitHub have other causes not work, like the need for Google Webcomponents that can be solved with Polly extension (https://github.com/WaterfoxCo/Waterfox-Classic/issues/41)

**Comment by WaywardHeart** [comment]
I looked into it, and the freeze is being caused by a mutation observer getting caught in an infinite loop. It looks like [Bug 1428246](https://bugzilla.mozilla.org/show_bug.cgi?id=1428246) should fix it.

**Comment by rami123** [comment]
> I looked into it, and the freeze is being caused by a mutation observer getting caught in an infinite loop. It looks like [Bug 1428246](https://bugzilla.mozilla.org/show_bug.cgi?id=1428246) should fix it.

Thanks

**Comment by Squall-Leonhart** [comment]
> I looked into it, and the freeze is being caused by a mutation observer getting caught in an infinite loop. It looks like [Bug 1428246](https://bugzilla.mozilla.org/show_bug.cgi?id=1428246) should fix it.

this looks like a right pain in the ass to bring over.

> This says that every java script that run on chaturbate.com and is originated from [www.gstatic.com](http://www.gstatic.com) will be blocked. (Important tag at the end gives this rule priority over other rules)

unfortunately not a great idea as this nasty site does some whacky things that leak memory if any scripts fail to load.

https://webcompat.com/issues/115500
